File: src/main/java/com/trading/kalyani/KPN/model/AutoTradeParams.java
package com.trading.kalyani.KPN.model;


import com.trading.kalyani.KPN.entity.JobIterationDetails;
import jakarta.persistence.*;
import lombok.*;

import java.time.LocalDateTime;

@Getter
@Setter
@ToString
@AllArgsConstructor
@NoArgsConstructor
@Entity
@Table(name = "AutoTradeParams")
public class AutoTradeParams {

    @Id
    @GeneratedValue(strategy= GenerationType.AUTO)
    private Long id;

    private boolean autoTradeFlag;
    private boolean closeTradeFlag;
    private String autoTradeCallSymbol;
    private Long autoTradeCallInstrumentToken;
    private Integer autoTradeCallLotSize;
    private Integer autoTradeCallEntryPrice;
    private Integer autoTradeCallSLPrice;
    private String autoTradePutSymbol;
    private Long autoTradePutInstrumentToken;
    private Integer autoTradePutLotSize;
    private Integer autoTradePutEntryPrice;
    private Integer autoTradePutSLPrice;
    private LocalDateTime autoTradeParamsTS;

    @OneToOne
    @JoinColumn(name = "jobIterationId", referencedColumnName = "id", nullable = false)
    private JobIterationDetails jobIterationDetails;


}

File: kpn-ui/src/app/models/autoTradeParams.ts
export class AutoTradeParams {
    autoTradeFlag: boolean;
    closeTradeFlag: boolean;
    autoTradeCallSymbol: string;
    autoTradeCallLotSize: number;
    autoTradeCallEntryDelta: number;
    autoTradeCallSLDelta: number;
    autoTradePutSymbol: string;
    autoTradePutLotSize: number;
    autoTradePutEntryDelta: number;
    autoTradePutSLDelta: number;

    constructor(
        autoTradeFlag: boolean = false,
        closeTradeFlag: boolean = false,
        autoTradeCallSymbol: string = '',
        autoTradeCallLotSize: number = 0,
        autoTradeCallEntryDelta: number = 0,
        autoTradeCallSLDelta: number = 0,
        autoTradePutSymbol: string = '',
        autoTradePutLotSize: number = 0,
        autoTradePutEntryDelta: number = 0,
        autoTradePutSLDelta: number = 0
    ) {
        this.autoTradeFlag = autoTradeFlag;
        this.closeTradeFlag = closeTradeFlag;
        this.autoTradeCallSymbol = autoTradeCallSymbol;
        this.autoTradeCallLotSize = autoTradeCallLotSize;
        this.autoTradeCallEntryDelta = autoTradeCallEntryDelta;
        this.autoTradeCallSLDelta = autoTradeCallSLDelta;
        this.autoTradePutSymbol = autoTradePutSymbol;
        this.autoTradePutLotSize = autoTradePutLotSize;
        this.autoTradePutEntryDelta = autoTradePutEntryDelta;
        this.autoTradePutSLDelta = autoTradePutSLDelta;
    }
}